addons/calendar/models/calendar_contact.py

```diff
@@ -19,3 +21,23 @@ class Contacts(models.Model):
     @api.model
     def unlink_from_partner_id(self, partner_id):
         return self.search([('partner_id', '=', partner_id)]).unlink()
+
+    @api.model
+    def update_check_filter(self, data):
```

Comment:
Could we have something a bit more "standard" than just receiving "data"?
Like directly having "partner_id" and "is_partner_checked" in the parameters?

And avoid calling the method with "all" as a partner_id from the JS?
The model business should be as clean as possible.

Or even better, could we find the associated "calendar.contacts" record ID from the JS and simply call the "write" function to set that field to True/False?

Thanks for having a look :)

Reply:
done, I like the last proposition :)